### PR TITLE
mmcls -> mmpretrain

### PR DESCRIPTION
**Pull request by huayuan4396**
## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

mmcls -> mmpretrain

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -97,6 +97,20 @@ jobs:
           ls -lah coverage.info
           cp coverage.info ../
 
+  badge_build_cpu_sdk:
```
Comment (RunningLeon) [inline .github/workflows/build.yml:100]: this should be removed from this pr and put in another PR

Comment (huayuan4396) [reply .github/workflows/build.yml:100]: OK, i will handle that later.

**Comment by RunningLeon** [comment]
@huayuan4396 Could sync with main  branch and fix conficts at first

File: requirements/codebases.txt [hunk 2]
```diff
@@ -1,8 +1,8 @@
-mmcls>=1.0.0rc2
 mmdet>=3.0.0
 mmdet3d>=1.1.0rc2
 mmedit>=1.0.0rc7
 mmocr>=1.0.0
 mmpose>=1.0.0
+mmpretrain>=1.0.0rc5
```
Comment (RunningLeon) [inline requirements/codebases.txt:6]: the first released verion of mmpretrain is `1.0.0rc7`

File: requirements/runtime.txt [hunk 3]
```diff
@@ -9,3 +9,4 @@ prettytable
 protobuf<=3.20.2
 six
 terminaltables
+transformers
```
Comment (RunningLeon) [inline requirements/runtime.txt:12]: should not add this 3rd party package.

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -281,7 +281,7 @@ jobs:
           python -m pip install openmim
           python -m pip install -r requirements.txt
           python -m pip install -r requirements/backends.txt
-          python -m mim install "mmcls>=1.0.0rc2"
+          python -m mim install "mmpretrain>=1.0.0rc5"
```
Comment (RunningLeon) [inline .github/workflows/build.yml:284]: rc7

File: docs/en/01-how-to-build/snpe.md [hunk 5]
```diff
@@ -71,18 +71,18 @@ $ python3 -m pip install -e .
 
 ## 3. Test the model
 
-Take Resnet-18 as an example. First refer to [documentation to install mmcls](https://github.com/open-mmlab/mmclassification/tree/1.x)  and  use `tools/deploy.py` to convert the model.
+Take Resnet-18 as an example. First refer to [documentation to install mmpretrain](https://github.com/open-mmlab/mmpretrain/tree/1.x)  and  use `tools/deploy.py` to convert the model.
```
Comment (RunningLeon) [inline docs/en/01-how-to-build/snpe.md:74]: the url is not right, should be on `main` branch

File: docs/en/02-how-to-run/convert_model.md [hunk 6]
```diff
@@ -4,14 +4,14 @@ This tutorial briefly introduces how to export an OpenMMlab model to a specific
...
-2. Install and build your target codebase. You could refer to [MMClassification-install](https://github.com/open-mmlab/mmclassification/blob/1.x/docs/en/get_started.md#installation), [MMDetection-install](https://github.com/open-mmlab/mmdetection/blob/3.x/docs/en/get_started.md), [MMSegmentation-install](https://github.com/open-mmlab/mmsegmentation/blob/1.x/docs/en/get_started.md#installation), [MMOCR-install](https://github.com/open-mmlab/mmocr/blob/1.x/docs/en/get_started/install.md), [MMEditing-install](https://github.com/open-mmlab/mmediting/blob/1.x/docs/en/get_started/install.md).
+2. Install and build your target codebase. You could refer to [MMPretrain-install](https://github.com/open-mmlab/mmpretrain/blob/1.x/docs/en/get_started.md#installation), [MMDetection-install](https://github.com/open-mmlab/mmdetection/blob/3.x/docs/en/get_started.md), [MMSegmentation-install](https://github.com/open-mmlab/mmsegmentation/blob/1.x/docs/en/get_started.md#installation), [MMOCR-install](https://github.com/open-mmlab/mmocr/blob/1.x/docs/en/get_started/install.md), [MMEditing-install](https://github.com/open-mmlab/mmediting/blob/1.x/docs/en/get_started/install.md).
```
Comment (RunningLeon) [inline docs/en/02-how-to-run/convert_model.md:14]: url is not right.

Comment (RunningLeon) [inline docs/en/02-how-to-run/convert_model.md:14]: url should be updated to main branch

File: docs/en/02-how-to-run/prebuilt_package_windows.md [hunk 7]
```diff
@@ -53,23 +53,23 @@ In order to use the prebuilt package, you need to install some third-party depen
...
-   cd mmclassification
+   git clone -b main https://github.com/open-mmlab/mmpretrain.git
+   cd mmpretrain
    pip install -e .
    ```
 
 4. Prepare a PyTorch model as our example
 
-   Download the pth [resnet18_8xb32_in1k_20210831-fbbb1da6.pth](https://download.openmmlab.com/mmclassification/v0/resnet/resnet18_8xb32_in1k_20210831-fbbb1da6.pth). The corresponding config of the model is [resnet18_8xb32_in1k.py](https://github.com/open-mmlab/mmclassification/blob/1.x/configs/resnet/resnet18_8xb32_in1k.py)
+   Download the pth [resnet18_8xb32_in1k_20210831-fbbb1da6.pth](https://download.openmmlab.com/mmclassification/v0/resnet/resnet18_8xb32_in1k_20210831-fbbb1da6.pth). The corresponding config of the model is [resnet18_8xb32_in1k.py](https://github.com/open-mmlab/mmpretrain/blob/1.x/configs/resnet/resnet18_8xb32_in1k.py)
```
Comment (RunningLeon) [inline docs/en/02-how-to-run/prebuilt_package_windows.md:66]: The config url should be updated to main branch

File: docs/en/02-how-to-run/write_config.md [hunk 8]
```diff
@@ -177,4 +177,4 @@ detection_tensorrt-int8_dynamic-320x320-1344x1344.py
...
+According to model's codebase, write the model config file. Model's config file is used to initialize the model, referring to [MMPretrain](https://github.com/open-mmlab/mmpretrain/blob/1.x/docs/en/user_guides/config.md), [MMDetection](https://github.com/open-mmlab/mmdetection/blob/3.x/docs/en/user_guides/config.md), [MMSegmentation](https://github.com/open-mmlab/mmsegmentation/blob/1.x/docs/en/user_guides/1_config.md), [MMOCR](https://github.com/open-mmlab/mmocr/blob/1.x/docs/en/user_guides/config.md), [MMEditing](https://github.com/open-mmlab/mmediting/blob/1.x/docs/en/user_guides/config.md).
```
Comment (RunningLeon) [inline docs/en/02-how-to-run/write_config.md:180]: url should be updated to main branch

Comment (RunningLeon) [inline docs/en/02-how-to-run/write_config.md:180]: url should be updated to main branch

File: docs/en/03-benchmark/benchmark.md [hunk 9]
```diff
@@ -64,7 +64,7 @@ Users can directly test the speed through [model profiling](../02-how-to-run/pro
     <td align="center" colspan="1">fp32</td>
   </tr>
   <tr>
-    <td align="center"><a href="https://github.com/open-mmlab/mmclassification/blob/1.x/configs/resnet/resnet50_8xb32_in1k.py"> ResNet </a></td>
+    <td align="center"><a href="https://github.com/open-mmlab/mmpretrain/blob/1.x/configs/resnet/resnet50_8xb32_in1k.py"> ResNet </a></td>
```
Comment (RunningLeon) [inline docs/en/03-benchmark/benchmark.md:67]: url should be updated to main branch

Comment (huayuan4396) [reply docs/en/03-benchmark/benchmark.md:67]: I will fix such kind of errors.

File: README.md [hunk 10]
```diff
@@ -172,7 +172,7 @@ This project is released under the [Apache 2.0 license](LICENSE).
 
 - [MMCV](https://github.com/open-mmlab/mmcv): OpenMMLab foundational library for computer vision.
 - [MIM](https://github.com/open-mmlab/mim): MIM installs OpenMMLab packages.
-- [MMClassification](https://github.com/open-mmlab/mmclassification): OpenMMLab image classification toolbox and benchmark.
+- [MMPretrain](https://github.com/open-mmlab/mmpretrain): OpenMMLab image classification toolbox and benchmark.
```
Comment (RunningLeon) [inline README.md:175]: `OpenMMLab pre-training toolbox and benchmark`

File: README_zh-CN.md [hunk 11]
```diff
@@ -162,7 +162,7 @@ MMDeploy 是 [OpenMMLab](https://openmmlab.com/) 模型部署工具箱，**为
 
 - [MMCV](https://github.com/open-mmlab/mmcv): OpenMMLab 计算机视觉基础库
 - [MIM](https://github.com/open-mmlab/mim): MIM 是 OpenMMlab 项目、算法、模型的统一入口
-- [MMClassification](https://github.com/open-mmlab/mmclassification): OpenMMLab 图像分类工具箱
+- [MMPretrain](https://github.com/open-mmlab/mmpretrain): OpenMMLab 图像分类工具箱
```
Comment (RunningLeon) [inline README_zh-CN.md:165]: OpenMMLab 深度学习预训练工具箱

File: docs/en/01-how-to-build/cmake_option.md [hunk 12]
```diff
@@ -121,9 +121,9 @@
   </tr>
   <tr>
     <td>MMDEPLOY_CODEBASES</td>
-    <td>{"mmcls", "mmdet", "mmseg", "mmedit", "mmocr", "all"}</td>
+    <td>{"mmpretrain", "mmdet", "mmseg", "mmedit", "mmocr", "all"}</td>
```
Comment (RunningLeon) [inline docs/en/01-how-to-build/cmake_option.md:124]: maybe we could map `mmretrain` to `mmcls` in sdk

**Comment by RunningLeon** [comment]
@huayuan4396 could fix conflicts

**Comment by irexyc** [comment]
There are some files that maybe need to rename mmcls to mmpretrain.
```
docs/en/04-supported-codebases/mmpretrain.md
docs/zh_cn/04-supported-codebases/mmpretrain.md
docs/zh_cn/07-developer-guide/support_new_model.md
docs/zh_cn/01-how-to-build/cmake_option.md
```

some files that maybe need to rename mmclassification to mmpretrain
```
docs/en/03-benchmark/supported_models.md
docs/zh_cn/03-benchmark/supported_models.md
docs/zh_cn/02-how-to-run/prebuilt_package_windows.md
.github/scripts/quantize_to_ncnn.py
.github/scripts/test_onnx2ncnn.py
```

**Comment by huayuan4396** [comment]
Some md link would be dead since the mmpretrain file path has not been merged.

**Comment by RunningLeon** [comment]
> Some md link would be dead since the mmpretrain file path has not been merged.

we could ignore lint error for doc links

**Review by irexyc — APPROVED** [review]
LGTM

**Review by RunningLeon — APPROVED** [review]
LGTM

**Comment by donatocayurin** [comment]
Thank you ❤️🙏
